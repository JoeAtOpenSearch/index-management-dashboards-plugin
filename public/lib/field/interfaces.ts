import React from "react";

export type FieldOption = {
  /**
   * All component changes will arrive here [set value will not trigger this function]
   */
  onChange?: (name: string, value?: any) => void;

  /**
   * Initialization data
   */
  values?: {};

  unmountComponent?: boolean;
};

export type ValidateResults = {
  errors: Record<string, string[]> | null;
  values: any;
};

export type InitResult<T> = {
  value?: T;
  onChange(value: T): void;
  ref?: React.RefCallback<any>;
};

export type Rule = {
  /**
   * cannot be empty (cannot be used with pattern)
   * @default true
   */
  required?: boolean;

  /**
   * error message
   */
  message?: string;

  /**
   * Check Regular Expression
   */
  pattern?: RegExp;
  /**
   * Minimum string length /minimum number of arrays
   */
  minLength?: number;
  /**
   * Maximum string length /maximum number of arrays
   */
  maxLength?: number;

  /**
   * String exact length /array exact number
   */
  length?: number;

  /**
   * minimum
   */
  min?: number;

  /**
   * maximum value
   */
  max?: number;
  /**
   * Summary of common patterns
   */
  format?: "url" | "email" | "tel" | "number";

  /**
   * Custom verification, (don't forget to execute callback() when the verification is successful, otherwise the verification will not return)
   */
  validator?: (rule: Rule, value: string | number | object | boolean | Date | null | any) => string | Promise<string>;

  /**
   * The name of the event that triggered the validation
   */
  trigger?: "onChange" | "onBlur" | string;
};

export type InitOption = {
  /**
   * The name of the field
   */
  name: string;

  /**
   * The name of the event that triggered the data change
   * @default 'onChange'
   */
  trigger?: string | "onChange" | "onBlur";

  /**
   * Check rules
   */
  rules?: Rule[];

  /**
   * Component custom events can be written here, others will be transparently transmitted (small package version ^0.3.0 support, large package ^0.7.0 support)
   */
  props?: any;
};

export type FieldInstance = {
  /**
   * Initialize each component
   */
  registerField(option?: InitOption): InitResult<any>;

  /**
   * check
   * @param names
   */
  validatePromise(names?: string[] | string): Promise<ValidateResults>;

  /**
   * 	Get the value of a single input control
   * @param field name
   */
  getValue(name: string): any;

  /**
   * Get the values ​​of a set of input controls, if no parameters are passed in, get the values ​​of all components
   * @param names
   */
  getValues(names?: string[]): any;

  /**
   * Set the value of a single input control (will trigger render, please follow the timing of react)
   */
  setValue(name: string, value: any): void;

  /**
   * Set the value of a set of input controls (will trigger render, please follow the timing of react)
   */
  setValues(obj: any): void;

  /**
   * Reset values
   */
  resetValues(obj: any): void;

  /**
   * Get the Error of a single input control
   */
  getError(name: string): null | string[];

  /**
   * Get the Error for a set of input controls
   * @param names field name
   */
  getErrors(names?: string[]): any;

  /**
   * Sets the Error for a single input control
   * @param name
   * @param errors
   */
  setError(name: string, errors: null | string[]): void;

  /**
   * Sets the Error for a set of input controls
   */
  setErrors(obj: any): void;
};

export type ValidateFunction = (
  rule: Rule & { field: string; aliasName?: string },
  value: string | number | object | boolean | Date
) => string | Promise<string> | undefined;
