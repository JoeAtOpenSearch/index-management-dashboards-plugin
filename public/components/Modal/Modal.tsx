/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiModal, EuiModalHeader, EuiModalHeaderTitle, EuiButton, EuiModalBody, EuiModalFooter } from "@elastic/eui";
import React, { Component, createContext, useState } from "react";
import { render } from "react-dom";

const ModalContext = createContext({
  component: null,
  props: {},
  onShow: (component: any, props: object) => {},
  onClose: () => {},
});

const ModalConsumer = ModalContext.Consumer;

class ModalProvider extends Component {
  state = { component: null, props: {} };

  onShow = (component: any, props: object): void => {
    this.setState({
      component,
      props,
    });
  };

  onClose = (): void => {
    this.setState({
      component: null,
      props: {},
    });
  };

  render() {
    return (
      <ModalContext.Provider value={{ ...this.state, onShow: this.onShow, onClose: this.onClose }}>
        {this.props.children}
      </ModalContext.Provider>
    );
  }
}

interface IShowOptions {
  title?: React.ReactChild;
  content?: React.ReactChild;
  type?: "alert" | "confirm";
  onOk?: () => void | Promise<any>;
  onCancel?: () => void | Promise<any>;
  onClose?: () => void;
  locale?: Partial<{
    ok: string;
    confirm: string;
    cancel: string;
  }>;
}

const blank = () => null;

const ModalApp = (props: IShowOptions) => {
  const { title, content, locale, onOk = blank, onCancel = blank, onClose = blank } = props;
  const defaultLocale: IShowOptions["locale"] = {
    ok: "OK",
    confirm: "Confirm",
    cancel: "Cancel",
  };
  const finalLocale: IShowOptions["locale"] = {
    ...defaultLocale,
    ...locale,
  };
  const [modalVisible, setModalVisible] = useState(true);
  const close = () => {
    setModalVisible(false);
    onClose();
  };
  return modalVisible ? (
    <EuiModal onClose={close}>
      <EuiModalHeader>
        <EuiModalHeaderTitle>
          <h1>{title}</h1>
        </EuiModalHeaderTitle>
      </EuiModalHeader>
      <EuiModalBody>{content}</EuiModalBody>
      <EuiModalFooter>
        {props.type === "confirm" ? (
          <>
            <EuiButton
              fill
              color="primary"
              onClick={async () => {
                await onOk();
                close();
              }}
            >
              {finalLocale.confirm}
            </EuiButton>
            <EuiButton
              fill
              color="secondary"
              onClick={async () => {
                await onCancel();
                close();
              }}
            >
              {finalLocale.cancel}
            </EuiButton>
          </>
        ) : (
          <>
            <EuiButton
              fill
              color="primary"
              onClick={async () => {
                await onOk();
                close();
              }}
            >
              {finalLocale.ok}
            </EuiButton>
          </>
        )}
      </EuiModalFooter>
    </EuiModal>
  ) : null;
};

const Modal = {
  show: (props: IShowOptions) => {
    const dom = document.createElement("div");
    document.body.appendChild(dom);
    const close = () => {
      dom.remove();
    };

    render(<ModalApp {...props} onClose={close} />, dom);
  },
};

export { ModalConsumer, ModalProvider, Modal };
