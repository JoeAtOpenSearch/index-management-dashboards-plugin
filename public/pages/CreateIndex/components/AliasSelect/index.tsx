/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { forwardRef } from "react";
import { EuiComboBoxProps } from "@elastic/eui";
import RemoteSelect, { RemoteSelectProps } from "../../../../components/RemoteSelect";
import { ServerResponse } from "../../../../../server/models/types";

export interface AliasSelectProps extends Omit<EuiComboBoxProps<{ label: string; value: string }>, "value" | "onChange"> {
  value?: Record<string, {}>;
  onChange?: (value: AliasSelectProps["value"]) => void;
  refreshOptions: (aliasName: string) => Promise<ServerResponse<{ alias: string }[]>>;
}

const transformObjToArray = (obj: AliasSelectProps["value"]): { label: string }[] => {
  return Object.keys(obj || {}).map((label) => ({ label }));
};
const transformArrayToObj = (array: { label: string }[]): AliasSelectProps["value"] => {
  return array.reduce((total, current) => ({ ...total, [current.label]: {} }), {});
};

const AliasSelect = forwardRef((props: AliasSelectProps, ref: React.Ref<HTMLInputElement>) => {
  const { value, onChange, refreshOptions: refreshOptionsFromProps } = props;
  const refreshOptions: RemoteSelectProps["refreshOptions"] = ({ searchValue }) => {
    return refreshOptionsFromProps(searchValue || "").then((res) => {
      if (res.ok) {
        return {
          ...res,
          response: [...new Set(res.response.map((item) => item.alias).filter((item) => !item.startsWith(".")))].map((item) => ({
            label: item,
          })),
        };
      } else {
        return res;
      }
    });
  };
  return (
    <RemoteSelect
      isDisabled={props.isDisabled}
      placeholder="Select or create aliases"
      customOptionText="Add {searchValue} as a new alias"
      refreshOptions={refreshOptions}
      value={transformObjToArray(value).map((item) => item.label)}
      onChange={(val) => {
        onChange && onChange(transformArrayToObj(val.map((label) => ({ label }))));
      }}
    />
  );
});

AliasSelect.displayName = "AliasSelect";

// @ts-ignore
export default AliasSelect;
