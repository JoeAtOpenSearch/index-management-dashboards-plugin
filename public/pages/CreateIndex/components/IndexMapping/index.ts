/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IndexMapping from "./IndexMapping";

export default IndexMapping;
