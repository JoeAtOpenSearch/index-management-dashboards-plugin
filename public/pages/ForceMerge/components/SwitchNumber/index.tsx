import React, { useState } from "react";
import { EuiRadioGroup, EuiSpacer } from "@elastic/eui";
import CustomFormRow from "../../../../components/CustomFormRow";
import { AllBuiltInComponents } from "../../../../components/FormGenerator";

interface SwitchNumberProps {
  value?: number;
  onChange: (val: SwitchNumberProps["value"]) => void;
}

export default function SwitchNumber(props: SwitchNumberProps) {
  const [id, setId] = useState(props.value && props.value > 0 ? "1" : "0");
  return (
    <>
      <EuiRadioGroup
        options={[
          {
            id: "0",
            label: "Automatically",
          },
          {
            id: "1",
            label: "Manually set",
          },
        ]}
        idSelected={id}
        onChange={(id) => {
          setId(id);
          if (id === "0") {
            props.onChange(undefined);
          } else {
            props.onChange(1);
          }
        }}
      />
      {id === "1" ? (
        <>
          <EuiSpacer size="s" />
          <CustomFormRow helpText="Specify the number of segments to merge to">
            <AllBuiltInComponents.Number {...props} />
          </CustomFormRow>
        </>
      ) : null}
    </>
  );
}
