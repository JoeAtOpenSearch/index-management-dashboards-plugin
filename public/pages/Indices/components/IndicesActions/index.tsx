/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React, { useCallback, useContext, useMemo, useState } from "react";
import {
  // @ts-ignore
  Criteria,
  // @ts-ignore
  Pagination,
  EuiButton,
  EuiContextMenu,
} from "@elastic/eui";

import { ManagedCatIndex } from "../../../../../server/models/interfaces";
import { ServerResponse } from "../../../../../server/models/types";
import ApplyPolicyModal from "../../components/ApplyPolicyModal";
import SimplePopover from "../../../../components/SimplePopover";
import { ModalConsumer } from "../../../../components/Modal";
import { CoreServicesContext } from "../../../../components/core_services";
import DeleteIndexModal from "../../components/DeleteIndexModal";
import { ServicesContext } from "../../../../services";

export interface IndicesActionsProps {
  selectedItems: ManagedCatIndex[];
  onDelete: (indices: string) => Promise<ServerResponse<any>>;
}

export default function IndicesActions(props: IndicesActionsProps) {
  const { selectedItems, onDelete } = props;
  const [deleteIndexModalVisible, setDeleteIndexModalVisible] = useState(false);
  const context = useContext(CoreServicesContext);
  const services = useContext(ServicesContext);

  const onDeleteIndexModalClose = () => {
    setDeleteIndexModalVisible(false);
  };

  const onDeleteIndexModalConfirm = useCallback(async () => {
    const result = await onDelete(selectedItems.map((item) => item.index).join(","));
    if (result && result.ok) {
      onDeleteIndexModalClose();
    }
  }, [selectedItems, onDelete, services, context, onDeleteIndexModalClose]);

  const renderKey = useMemo(() => Date.now(), [selectedItems]);

  return (
    <>
      <ModalConsumer>
        {({ onShow }) => (
          <SimplePopover
            data-test-subj="More Action"
            panelPaddingSize="none"
            button={
              <EuiButton iconType="arrowDown" iconSide="right">
                Actions
              </EuiButton>
            }
          >
            <EuiContextMenu
              initialPanelId={0}
              // The EuiContextMenu has bug when testing in jest
              // the props change won't make it rerender
              key={renderKey}
              panels={[
                {
                  id: 0,
                  items: [
                    {
                      name: "Delete",
                      disabled: !selectedItems.length,
                      "data-test-subj": "Delete Action",
                      onClick: () => setDeleteIndexModalVisible(true),
                    },
                    {
                      name: "Apply policy",
                      disabled: !selectedItems.length,
                      "data-test-subj": "Apply policyButton",
                      onClick: () =>
                        onShow(ApplyPolicyModal, {
                          indices: selectedItems.map((item: ManagedCatIndex) => item.index),
                          core: context,
                        }),
                    },
                  ],
                },
              ]}
            />
          </SimplePopover>
        )}
      </ModalConsumer>
      <DeleteIndexModal
        selectedItems={selectedItems.map((item) => item.index)}
        visible={deleteIndexModalVisible}
        onClose={onDeleteIndexModalClose}
        onConfirm={onDeleteIndexModalConfirm}
      />
    </>
  );
}
