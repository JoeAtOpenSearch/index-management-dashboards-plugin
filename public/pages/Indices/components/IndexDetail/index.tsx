/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState } from "react";
import { EuiButtonEmpty, EuiCopy } from "@elastic/eui";
import { IndexItem } from "../../../../../models/interfaces";
import { Modal } from "../../../../components/Modal";
import IndicesActions from "../IndicesActions";
import { ManagedCatIndex } from "../../../../../server/models/interfaces";

export interface IndexDetailModalProps {
  index: string;
  onDelete?: () => void;
  record: ManagedCatIndex;
}

export default function IndexDetail(props: IndexDetailModalProps) {
  // const { detail, visible } = props;
  const { index, record, onDelete } = props;
  const [visible, setVisible] = useState(false);
  const [detail, setDetail] = useState({} as IndexItem);
  return (
    <>
      <EuiCopy textToCopy={index}>
        {(copy) => <EuiButtonEmpty size="xs" flush="right" iconType="copyClipboard" onClick={copy} color="text"></EuiButtonEmpty>}
      </EuiCopy>
      <EuiButtonEmpty onClick={() => setVisible(true)} data-test-subj={`view-index-detail-button-${index}`}>
        {index}
      </EuiButtonEmpty>
      <Modal.SimpleModal
        visible={visible}
        maxWidth={false}
        style={{
          width: "60vw",
        }}
        onClose={() => setVisible(false)}
        title={
          <div style={{ display: "flex", justifyContent: "space-between" }}>
            <span>{index}</span>
            <IndicesActions selectedItems={[record]} onDelete={onDelete} />
          </div>
        }
      />
    </>
  );
}
