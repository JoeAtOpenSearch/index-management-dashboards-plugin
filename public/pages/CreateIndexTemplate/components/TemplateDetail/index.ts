/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IndexDetail from "./TemplateDetail";

export default IndexDetail;
export * from "./TemplateDetail";
