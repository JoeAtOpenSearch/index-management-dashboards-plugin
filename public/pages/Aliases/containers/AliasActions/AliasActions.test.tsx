/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, waitFor } from "@testing-library/react";
// @ts-ignore
import userEvent from "@testing-library/user-event";
import { browserServicesMock, coreServicesMock } from "../../../../../test/mocks";
import AliasesActions, { AliasesActionsProps } from "./index";
import { ModalProvider } from "../../../../components/Modal";
import { ServicesContext } from "../../../../services";
import { CoreServicesContext } from "../../../../components/core_services";
import { IAPICaller } from "../../../../../models/interfaces";

const exampleBlocksStateResponse = {
  cluster_name: "opensearch-cluster",
  cluster_uuid: "123",
  blocks: {
    indices: {
      test_index1: {
        "4": {
          description: "index closed",
          retryable: false,
          levels: ["read", "write"],
        },
      },
      test_index2: {
        "5": {
          description: "index read-only (api)",
          retryable: false,
          levels: ["write", "metadata_write"],
        },
      },
    },
  },
};

function renderWithRouter(props: Omit<AliasesActionsProps, "history">) {
  return {
    ...render(
      <CoreServicesContext.Provider value={coreServicesMock}>
        <ServicesContext.Provider value={browserServicesMock}>
          <ModalProvider>
            <AliasesActions {...props} history={{} as any} />
          </ModalProvider>
        </ServicesContext.Provider>
      </CoreServicesContext.Provider>
    ),
  };
}

describe("<AliasesActions /> spec", () => {
  it("renders the component and all the actions should be disabled when no items selected", async () => {
    const { container, getByTestId } = renderWithRouter({
      selectedItems: [],
      onUpdateAlias: () => null,
      onDelete: () => null,
    });

    await waitFor(() => {
      expect(container.firstChild).toMatchSnapshot();
    });

    userEvent.click(document.querySelector('[data-test-subj="moreAction"] button') as Element);
    await waitFor(() => {
      expect(getByTestId("deleteAction")).toBeDisabled();
    });
  });

  it("delete alias by calling commonService", async () => {
    const onDelete = jest.fn();
    let times = 0;
    browserServicesMock.commonService.apiCaller = jest.fn(
      async (payload): Promise<any> => {
        if (payload.endpoint === "indices.deleteAlias") {
          if (times >= 1) {
            return {
              ok: true,
              response: {},
            };
          } else {
            times++;
            return {
              ok: false,
              error: "test error",
            };
          }
        }
        return { ok: true, response: {} };
      }
    );
    const { container, getByTestId, getByPlaceholderText } = renderWithRouter({
      selectedItems: [
        {
          index: "test_index",
          alias: "1",
          filter: "1",
          "routing.index": "1",
          "routing.search": "1",
          is_write_index: "1",
          indexArray: ["test_index"],
        },
      ],
      onUpdateAlias: () => null,
      onDelete,
    });

    await waitFor(() => {
      expect(container.firstChild).toMatchSnapshot();
    });

    userEvent.click(document.querySelector('[data-test-subj="moreAction"] button') as Element);
    userEvent.click(getByTestId("deleteAction"));
    userEvent.type(getByPlaceholderText("delete"), "delete");
    userEvent.click(getByTestId("deleteConfirmButton"));

    await waitFor(() => {
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledTimes(1);
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledWith({
        endpoint: "indices.deleteAlias",
        data: {
          index: ["test_index"],
          name: ["1"],
        },
      });
      expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledTimes(1);
      expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledWith("test error");
      expect(onDelete).toHaveBeenCalledTimes(0);
    });

    userEvent.click(getByTestId("deleteConfirmButton"));

    await waitFor(() => {
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledTimes(2);
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledTimes(1);
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledWith("Delete [1] successfully");
      expect(onDelete).toHaveBeenCalledTimes(1);
    });
  });

  it("flush alias by calling commonService", async () => {
    browserServicesMock.commonService.apiCaller = jest.fn().mockImplementation((params: IAPICaller) => {
      if (params.endpoint === "indices.flush") {
        return { ok: true, response: {} };
      } else {
        return {
          ok: true,
          response: exampleBlocksStateResponse,
        };
      }
    });

    const { container, getByTestId } = renderWithRouter({
      selectedItems: [
        {
          alias: "alias1",
          indexArray: ["test_index1", "test_index3"],
        },
        {
          alias: "alias2",
          indexArray: ["test_index3"],
        },
      ],
    });

    userEvent.click(document.querySelector('[data-test-subj="moreAction"] button') as Element);
    userEvent.click(getByTestId("Flush Action"));
    await waitFor(() => {
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledTimes(1);
      expect(getByTestId("Flush Modal Title")).toHaveTextContent("Flush alias");
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledWith({
        endpoint: "cluster.state",
        data: {
          metric: "blocks",
        },
      });
    });
    userEvent.click(getByTestId("Flush Confirm button"));

    await waitFor(() => {
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledTimes(2);
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledWith({
        endpoint: "indices.flush",
        data: {
          index: "alias2",
        },
      });
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledTimes(1);
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledWith("Flush [alias2] successfully");
    });
  });
});
