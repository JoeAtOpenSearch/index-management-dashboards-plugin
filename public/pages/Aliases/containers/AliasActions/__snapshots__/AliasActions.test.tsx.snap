// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AliasesActions /> spec delete alias by calling commonService 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<AliasesActions /> spec refresh alias blocked because all index are blocked 1`] = `
<body
  class="euiBody-hasOverlayMask"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      class="euiPopover euiPopover--anchorDownCenter euiPopover-isOpen"
      data-test-subj="moreAction"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButton euiButton--primary"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButton__content"
          >
            EuiIconMock
            <span
              class="euiButton__text"
            >
              Actions
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="euiOverlayMask euiOverlayMask--aboveHeader"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="euiModal euiModal--maxWidth-default"
        tabindex="0"
      >
        <button
          aria-label="Closes this modal window"
          class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall euiModal__closeIcon"
          type="button"
        >
          EuiIconMock
        </button>
        <div
          class="euiModal__flex"
        >
          <div
            class="euiModalHeader"
          >
            <div
              class="euiModalHeader__title"
            >
              Refresh 
              aliases
            </div>
          </div>
          <div
            class="euiModalBody"
          >
            <div
              class="euiModalBody__overflow"
            >
              <div
                style="line-height: 1.5;"
              >
                <div
                  class="euiSpacer euiSpacer--l"
                />
                <div
                  class="euiCallOut euiCallOut--warning"
                >
                  <div
                    class="euiCallOutHeader"
                  >
                    EuiIconMock
                    <span
                      class="euiCallOutHeader__title"
                    >
                      The following aliases will not be refreshed because they are closed.
                    </span>
                  </div>
                </div>
                <p />
                <ul
                  style="list-style-type: disc; list-style-position: inside;"
                >
                  <li
                    data-test-subj="BlockedItem-1"
                  >
                    1
                  </li>
                  <li
                    data-test-subj="BlockedItem-2"
                  >
                    2
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="euiModalFooter"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--primary"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonEmpty__content"
              >
                <span
                  class="euiButtonEmpty__text"
                >
                  Cancel
                </span>
              </span>
            </button>
            <button
              class="euiButton euiButton--primary euiButton--fill euiButton-isDisabled"
              data-test-subj="refreshConfirmButton"
              disabled=""
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Refresh
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`<AliasesActions /> spec refresh alias by calling commonService 1`] = `
<body
  class="euiBody-hasOverlayMask"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      class="euiPopover euiPopover--anchorDownCenter euiPopover-isOpen"
      data-test-subj="moreAction"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButton euiButton--primary"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButton__content"
          >
            EuiIconMock
            <span
              class="euiButton__text"
            >
              Actions
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="euiOverlayMask euiOverlayMask--aboveHeader"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="euiModal euiModal--maxWidth-default"
        tabindex="0"
      >
        <button
          aria-label="Closes this modal window"
          class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall euiModal__closeIcon"
          type="button"
        >
          EuiIconMock
        </button>
        <div
          class="euiModal__flex"
        >
          <div
            class="euiModalHeader"
          >
            <div
              class="euiModalHeader__title"
            >
              Refresh 
              aliases
            </div>
          </div>
          <div
            class="euiModalBody"
          >
            <div
              class="euiModalBody__overflow"
            >
              <div
                style="line-height: 1.5;"
              >
                <p>
                  The following 
                  aliases
                   will be refreshed.
                </p>
                <ul
                  style="list-style-type: disc; list-style-position: inside;"
                >
                  <li
                    data-test-subj="UnblockedItem-2"
                  >
                    2
                  </li>
                </ul>
                <div
                  class="euiSpacer euiSpacer--l"
                />
                <div
                  class="euiCallOut euiCallOut--warning"
                >
                  <div
                    class="euiCallOutHeader"
                  >
                    EuiIconMock
                    <span
                      class="euiCallOutHeader__title"
                    >
                      The following aliases will not be refreshed because they are closed.
                    </span>
                  </div>
                </div>
                <p />
                <ul
                  style="list-style-type: disc; list-style-position: inside;"
                >
                  <li
                    data-test-subj="BlockedItem-1"
                  >
                    1
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="euiModalFooter"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--primary"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonEmpty__content"
              >
                <span
                  class="euiButtonEmpty__text"
                >
                  Cancel
                </span>
              </span>
            </button>
            <button
              class="euiButton euiButton--primary euiButton--fill"
              data-test-subj="refreshConfirmButton"
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Refresh
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`<AliasesActions /> spec refresh alias even failed to get index status 1`] = `
<body
  class="euiBody-hasOverlayMask"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      class="euiPopover euiPopover--anchorDownCenter euiPopover-isOpen"
      data-test-subj="moreAction"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButton euiButton--primary"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButton__content"
          >
            EuiIconMock
            <span
              class="euiButton__text"
            >
              Actions
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="euiOverlayMask euiOverlayMask--aboveHeader"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="euiModal euiModal--maxWidth-default"
        tabindex="0"
      >
        <button
          aria-label="Closes this modal window"
          class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall euiModal__closeIcon"
          type="button"
        >
          EuiIconMock
        </button>
        <div
          class="euiModal__flex"
        >
          <div
            class="euiModalHeader"
          >
            <div
              class="euiModalHeader__title"
            >
              Refresh 
              aliases
            </div>
          </div>
          <div
            class="euiModalBody"
          >
            <div
              class="euiModalBody__overflow"
            >
              <div
                style="line-height: 1.5;"
              >
                <p>
                  The following 
                  aliases
                   will be refreshed.
                </p>
                <ul
                  style="list-style-type: disc; list-style-position: inside;"
                >
                  <li
                    data-test-subj="UnblockedItem-1"
                  >
                    1
                  </li>
                  <li
                    data-test-subj="UnblockedItem-2"
                  >
                    2
                  </li>
                </ul>
                <div
                  class="euiSpacer euiSpacer--l"
                />
              </div>
            </div>
          </div>
          <div
            class="euiModalFooter"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--primary"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonEmpty__content"
              >
                <span
                  class="euiButtonEmpty__text"
                >
                  Cancel
                </span>
              </span>
            </button>
            <button
              class="euiButton euiButton--primary euiButton--fill"
              data-test-subj="refreshConfirmButton"
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Refresh
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`<AliasesActions /> spec renders the component and all the actions should be disabled when no items selected 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;
